ext/mini_racer_extension/mini_racer_extension.cc

```diff
@@ -225,7 +225,13 @@ static Handle<Value> convert_ruby_to_v8(Isolate* isolate, VALUE value) {
 
     switch (TYPE(value)) {
     case T_FIXNUM:
-	return scope.Escape(Integer::New(isolate, NUM2INT(value)));
+        length = NUM2INT(rb_funcall(value, rb_intern("size"), 0));
+        if (length > 4)
+        {
+            return scope.Escape(Number::New(isolate, NUM2DBL(value)));
```

Comment:
Also, I'm not sure where the code came from initially, and why it was not done that way to begin with? Also NUM2LONG exists, but doesn't work with the Integer class which is looking for an int32